### PR TITLE
 feat(jdbc-catalog): Support OceanBase JDBC Catalog

### LINKED ISSUE #4353: [FEATURE] Support OceanBase-CE JDBC Catalog
### Describe the feature

Support OceanBase as a type of data source.

### Motivation

[OceanBase](https://www.oceanbase.com/) is an enterprise-ready [open source](https://github.com/oceanbase/oceanbase) distributed relational database with high availability, high performance, horizontal scalability, and compatibility with SQL standards.
We should consider supporting OceanBase as a data source, including features for managing databases, tables etc.

### Describe the solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yuanoOo**
### What changes were proposed in this pull request?

This closes #4353 

### How was this patch tested?

UT & IT

### CODE REVIEW

**Comment by xunliu** [comment]
hi @yuanoOo Thank you for your contribution.

File: catalogs/catalog-jdbc-oceanbase/src/main/resources/jdbc-oceanbase.conf [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# jdbc-url = jdbc:mysql://localhost:2881/
+# jdbc-user = strato
+# jdbc-password = strato
```
Comment (justinmclean) [inline catalogs/catalog-jdbc-oceanbase/src/main/resources/jdbc-oceanbase.conf:21]: Please select a different name and password to be vendor neutral.

**Comment by yuanoOo** [comment]
@xunliu I just fixed failing test cases in integration tests, Please help trigger the github workflow again.

**Comment by yuqi1129** [comment]
> @xunliu I just fixed failing test cases in integration tests, Please help trigger the github workflow again.

done

**Review by mchades — COMMENTED** [review]
Could you please break this PR into smaller PRs as it is too big to review now

**Comment by yuanoOo** [comment]
> Could you please break this PR into smaller PRs as it is too big to review now

OK, I will break it and track it using EPIC issue in my free time.
Also, I have updated the design document . #4849

**Comment by jerryshao** [comment]
Hi @yuanoOo , since OB related PRs are already merged, I'm going to close this PR. Thanks a lot for your contributions and greatly appreciated.

**Pull request closed without merging**